conftest.py

```diff
@@ -1,10 +1,37 @@
 import os,pytest
+from page_objects.PageFactory import PageFactory
 from conf import browser_os_name_conf
 from utils import post_test_reports_to_slack
 from utils.email_pytest_report import Email_Pytest_Report
 from utils import Tesults
 
-
+@pytest.fixture
+def test_obj(base_url,browser,browser_version,os_version,os_name,remote_flag,testrail_flag,tesults_flag,test_run_id,remote_project_name,remote_build_name):
+    
+    "Return an instance of Base Page that knows about the third party integrations"
+    test_obj = PageFactory.get_page_object("Zero",base_url=base_url)
+
+    #Setup and register a driver
+    test_obj.register_driver(remote_flag,os_name,os_version,browser,browser_version,remote_project_name,remote_build_name)
+
+    #Setup TestRail reporting
+    if testrail_flag.lower()=='y':
+        if test_run_id is None:
+            test_obj.write('\033[91m'+"\n\nTestRail Integration Exception: It looks like you are trying to use TestRail Integration without providing test run id. \nPlease provide a valid test run id along with test run command using -R flag and try again. for eg: pytest -X Y -R 100\n"+'\033[0m')
+            testrail_flag = 'N'   
+        if test_run_id is not None:
+            test_obj.register_testrail()
+            test_obj.set_test_run_id(test_run_id)
+
+    if tesults_flag.lower()=='y':
+        test_obj.register_tesults()
+    
```

Comment:
@rohinigopalqxf2 - I am having a hard time understanding lines 29-33. Specifically:

a) why yield instead of return?
b) why is teardown() being forced? 

Since the tests run, I believe what you are doing is right. I just want a verbal clarification on how this works.

Reply:
Please find my replies below :

a) As per my understanding of the yield. The yield statement suspends execution of a function and sends a value back to the caller, while saving state and later resuming meaning the whole generator itself can still be resumed after the return value is obtained. 
And a return statement ends the execution of the function and sends a value back to the caller.

b) It was used in same way in each test, I just moved it to fixture.

Reply:
Ok, thanks!